### PR TITLE
add bookmarks support

### PR DESCRIPTION
**Pull request by PythonFZ**
With this PR we introduce `bookmarks` that allow to jump to specific scenes.
Bookmarks can be set by pressing `b` or through Python
```py
vis = ZnDraw(url="http://127.0.0.1:1234/")

print(vis.bookmarks) # dict

vis.bookmarks = {25: "Hello"} # add a bookmark for step 25 with the text "Hello"
```
Because of the way `property` works, it is **not** possible to use `vis.bookmarks.update({33: "something"})`!

This feature is also available in the new modifiers, allowing modifications to indicate what happend at the specific point in time.
Modifiers should always use `vis.bookmarks = vis.bookmarks | { <new bookmarks> }`

image

### CODE REVIEW

File: zndraw/static/pycom/Bookmarks.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      }.bind(this),
+    );
+    this.socket.on("bookmarks:set", (bookmarks) => {
+      console.log("bookmarks:set");
+      this.bookmarks = bookmarks;
+      this.updateBookmarks();
+    });
+
+    // on keypress b set the bookmark
+    window.addEventListener("keypress", (event) => {
```
Comment (RokasEl) [inline zndraw/static/pycom/Bookmarks.js:22]: Add shift-B to cycle through created bookmarks?

Comment (RokasEl) [reply zndraw/static/pycom/Bookmarks.js:22]: Also, how to delete bookmars?

**Review by RokasEl — CHANGES_REQUESTED** [review]
If I set a bookmark at step 100, and do a modifier at step 50 all structures past 50 get deleted. However, the bookmark remains. Maybe it should be deleted as well?

Also, this is a small thing, but when I create a bookmark at step 100 and then do a modification, the bookmark remains on the very far right side of the progress bar. Would it be very hard to update its position to reflect that step 100 is not the last step anymore?

File: zndraw/static/pycom/Bookmarks.js (hunk 1 above)
Comment (PythonFZ) [reply zndraw/static/pycom/Bookmarks.js:22]: I've added shift + arrow left/right to iterate and shift + click the bookmark to remove it.

**Pull request merged**